app/models/spree/order_contents.rb

```diff
@@ -38,6 +38,7 @@ def update_or_create(variant, attributes)
         line_item.price = variant.price
         order.line_items << line_item
       end
+      update_shipment
```

Comment:
Is it relevant for both create and update? I thought it might only need to be called on update.

Reply:
You need to update shipments in both scenario, because a customer could be either adding a new item in the cart or updating the quantity of an existing item. That said updating the shipment calls `order.ensure_updated_shipments`, which won't do anything if there is no existing shipment.
 
https://github.com/openfoodfoundation/openfoodnetwork/blob/4c71ea38667dcaa45ad354399ce22f500cb0bab1/app/models/spree/order.rb#L512-L517